pkg/operator/configobservation/apiserver/observe_termination_duration.go

```diff
@@ -0,0 +1,131 @@
+func ObserveShutdownDelayDuration(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains shutdown-delay-duration field.
+		ret = configobserver.Pruned(ret, shutdownDelayDurationPath)
+	}()
+	var currentShutdownDelayDuration string
+
+	// read the current value
+	currentShutdownDelaySlice, _, err := unstructured.NestedStringSlice(existingConfig, shutdownDelayDurationPath...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to extract shutdown delay duration from the existing config: %v", err))
+	}
+
+	if len(currentShutdownDelaySlice) > 0 {
+		currentShutdownDelayDuration = currentShutdownDelaySlice[0]
+	}
+
+	// read the observed value
+	var observedShutdownDelayDuration string
+	listers := genericListers.(configobservation.Listers)
+	infra, err := listers.InfrastructureLister().Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// without the infrastructure object we are not able to determine the type of platform we are running on
+		return existingConfig, append(errs, err)
+	}
+	switch infra.Spec.PlatformSpec.Type {
+	case configv1.AWSPlatformType:
+		// AWS has a known issue: https://bugzilla.redhat.com/show_bug.cgi?id=1943804
+		// We need to extend the shutdown-delay-duration so that an NLB has a chance to notice and remove unhealthy instance.
+		// Once the mentioned issue is resolved this code must be removed and default values applied
+		observedShutdownDelayDuration = "210s"
+	default:
+		observedShutdownDelayDuration = currentShutdownDelayDuration
+	}
+
+	// see if the current and the observed value differ
+	observedConfig := map[string]interface{}{}
+	if currentShutdownDelayDuration != observedShutdownDelayDuration {
+		if err = unstructured.SetNestedStringSlice(observedConfig, []string{observedShutdownDelayDuration}, shutdownDelayDurationPath...); err != nil {
+			return existingConfig, append(errs, err)
+		}
+
+		recorder.Eventf("ObserveShutdownDelayDuration", "shutdown-delay-duration changed to %q", observedShutdownDelayDuration)
+		return observedConfig, errs
+	}
+
+	// nothing has changed return the original configuration
+	return existingConfig, errs
+}
+
+// ObserveShutdownDelayDuration sets the watch termination duration according to the current platform.
+func ObserveWatchTerminationDuration(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains watchTerminationDuration field.
+		ret = configobserver.Pruned(ret, watchTerminationDurationPath)
+	}()
+	var currentWatchTerminationDuration string
+	var err error
+
+	// read the current value
+	currentWatchTerminationDuration, _, err = unstructured.NestedString(existingConfig, watchTerminationDurationPath...)
```

Comment:
:=

Reply:
why? I want to set `currentWatchTerminationDuration` and `err`